src/pytest_codspeed/plugin.py

```diff
@@ -318,14 +329,69 @@ def __init__(self, request: pytest.FixtureRequest):
         self.extra_info: dict = {}
 
         self._request = request
-
-    def __call__(self, func: Callable[P, T], *args: P.args, **kwargs: P.kwargs) -> T:
-        config = self._request.config
-        plugin = get_plugin(config)
-        if plugin.is_codspeed_enabled:
-            return _measure(plugin, self._request.node, config, func, *args, **kwargs)
+        self._config = self._request.config
+        self._plugin = get_plugin(self._config)
+        self._called = False
+
+    def __call__(
+        self, target: Callable[..., T], *args: tuple, **kwargs: dict[str, Any]
```

Comment:
this used to be correct, now it makes no sense at all.

what you changed it to is “each positional argument is a tuple, and each keyword argument is a dict”. So this is valid according to your typing:

```py
benchmark(print, (), (1,2), file={})
```

but this is invalid according to your typing:

```py
benchmark(print, "a", file="out.txt")
```